components/ai_chat/resources/page/state/conversation_context.tsx

```diff
@@ -547,6 +556,59 @@ export function ConversationContextProvider(props: React.PropsWithChildren) {
       })
     }
   }
+  const ignoreExternalLinkWarning = React.useRef(
+    JSON.parse(
+      localStorage.getItem(IGNORE_EXTERNAL_LINK_WARNING_KEY) ?? 'false'
+    )
+  )
```

Comment:
I'm afraid this is still causing `JSON.parse` and `localStorage.getItem` to fire on every re-render. The only difference is now the result isn't being used. It'll need a React.useMemo or even a module-level variable, outside of the context function.

Reply:
Fixed via https://github.com/brave/brave-core/compare/4da09bcf15482c11dc4b3893042444780727ac9c..d04fa40465679f5701e7765c8acee413abd0cfb8